### PR TITLE
Crosslink heartbeat verify through current committee

### PR DESCRIPTION
**Pull request by Frozen**
Crosslink heartbeat verify through current committee

### CODE REVIEW

**Comment by Frozen** [comment]
https://github.com/harmony-one/harmony/pull/4673/files#diff-53cafc20d348cad861dbaad811b8e24eed9320f4382536da667b166aa641eb6eR71

**Review by ONECasey — APPROVED** [review]

**Comment by sophoah** [comment]
PR Explanation:
In current dev shard 0 node are signing the crosslink heartbeat with the current leader private key, however upon processing the heartbeat shard 1 is verifying the public key by using the epoch number in the heartbeat hb.epoch(). In normal situation it usually works fine since hb.epoch() and cur.epoch() will be the same or more or less 1 epoch difference. However, in cases where the crosslink wasn't processed for a long time, the leader in hb.epoch() may not have been elected. It will be especially true in the future with external leader rotation when external validator can be kick out from committee due to low signature rate.

This PR is takes the assumption that the heartbeat has to be processed as soon as they are sent, and making sure it is sent by a valid/elected validator should be enough for now

File: node/node_cross_link.go [hunk 1]
```diff
@@ -103,7 +103,12 @@ func (node *Node) processCrossLinkHeartbeatMessage(msgPayload []byte) error {
 func (node *Node) ProcessCrossLinkMessage(msgPayload []byte) {
 	if node.IsRunningBeaconChain() {
 		pendingCLs, err := node.Blockchain().ReadPendingCrossLinks()
-		if err == nil && len(pendingCLs) >= maxPendingCrossLinkSize {
+		if err != nil {
+			utils.Logger().Debug().
```
Comment (GheisMohammadi) [inline node/node_cross_link.go:107]: it would be good to have the error in logs as well
```golang
utils.Logger().Debug().Err(err).
```

**Comment by sophoah** [comment]
@Frozen can you review the conflict and @GheisMohammadi comment

File: node/node_cross_link.go (hunk 1 above)
Comment (Frozen) [reply node/node_cross_link.go:107]: Added log message.

**Review by GheisMohammadi — APPROVED** [review]

**Pull request merged**